modular_skyrat/modules/modular_items/lewd_items/code/lewd_arousal/arousal.dm

```diff
@@ -14,15 +14,13 @@
 			arousal_status = arousal_flag
 			if(istype(src, /mob/living/carbon/human))
 				var/mob/living/carbon/human/target = src
-				for(var/i = 1, i <= target.external_organs.len, i++)
-					if(istype(target.external_organs[i], /obj/item/organ/external/genital))
-						var/obj/item/organ/external/genital/target_genital = target.external_organs[i]
-						if(!target_genital.aroused == AROUSAL_CANT)
-							target_genital.aroused = arousal_status
-							target_genital.update_sprite_suffix()
+				for(var/obj/item/organ/external/genital/target_genital in target.external_organs)
+					if(!target_genital.aroused == AROUSAL_CANT)
+						target_genital.aroused = arousal_status
+						target_genital.update_sprite_suffix()
 				target.update_body()
 	else
 		arousal -= abs(arous)
 
-		arousal = min(max(arousal, 0), 100)
+		arousal = clamp(arousal, 0, AROUSAL_LIMIT)
```

Comment:
You could put this all on one line by making it into `arousal = clamp(arousal - abs(arous), 0, AROUSAL_LIMIT)` (thanks GitHub for not letting me put this up as a suggestion >:( )